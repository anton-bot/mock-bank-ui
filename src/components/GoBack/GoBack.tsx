import { Link } from 'react-router-dom';
import './GoBack.scss';

type Props = {
  to: string;
};

export const GoBack: React.FC<Props> = (props) => {
  return (
    <div className="GoBack">
      <Link to={props.to}>{props.children}</Link>
    </div>
  );
};
