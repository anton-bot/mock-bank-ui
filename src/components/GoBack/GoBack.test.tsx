import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { GoBack } from './GoBack';
import renderer from 'react-test-renderer';

describe('Go back link', () => {
  it('must render a link', () => {
    const { getByText } = render(
      <BrowserRouter>
        <GoBack to="/test">Go back to main page</GoBack>
      </BrowserRouter>,
    );

    expect(getByText(/Go back to main page/)).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <GoBack to="/test">Go back to main page</GoBack>
        </BrowserRouter>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
