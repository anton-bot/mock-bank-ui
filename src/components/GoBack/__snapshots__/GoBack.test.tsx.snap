// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Go back link matches snapshot 1`] = `
<div
  className="GoBack"
>
  <a
    href="/test"
    onClick={[Function]}
  >
    Go back to main page
  </a>
</div>
`;
